spec/Section 3 -- Type System.md

```diff
@@ -287,6 +288,11 @@ A `Union` defines a list of possible types; similar to interfaces, whenever the
 type system claims a union will be returned, one of the possible types will be
 returned.
 
+An `Intersection` defines a list of constraining abstract types. If a field
+claims it returns an Intersection type, it will return only types that are
+contained within all of the Intersections's Unions and types that implement all
+of the Intersection's Interfaces.
+
```

Comment:
1. what is Intersection's Unions? in the grammar rule you mention IntersectionMemberTypes
2. so... if at runtime field resolver returns a value of type that IS one of union types but does not implement one of interfaces - then what? it is replaced with null? what if field is not nullable? Hell confusing

Reply:
At runtime, only concrete object types can be returned as the actual type just like with any other abstract type.

Reply:
And yes, an intersection's unions are its member types that are unions. I copied the grammar from unions where even though object types are the only currently allowed member types, that is not specified at the grammar level, it's specified only as part of validation.

Reply:
If a non possible type is returned at runtime you get the same error as if you return a non implementing type when using an interface or an object type not in the Union.

In the implementation changes, there was no need to directly touch Execution because the only thing that changes is what goes on in the isSubType method